kevm-pyk/src/kevm_pyk/kevm.py

```diff
@@ -378,11 +378,17 @@ def _replace(term: KInner) -> KInner:
             if type(term) is KApply and term.label.name == '<gas>':
                 gas_term = term.args[0]
                 if type(gas_term) is KApply and gas_term.label.name == 'infGas':
-                    result = KApply('<gas>', KApply('infGas', abstract_term_safely(term)))
+                    if type(gas_term.args[0]) is KVariable:
+                        return term
+                    result = KApply(
+                        '<gas>', KApply('infGas', abstract_term_safely(term, base_name='VGAS', sort=KSort('Int')))
+                    )
                     return result
```

Comment:
```suggestion
                    return KApply(
                        '<gas>', KApply('infGas', abstract_term_safely(term, base_name='VGAS', sort=KSort('Int')))
                    )
```

Reply:
Also, you can use `INT` instead of `KSort('Int')` if you do `from pyk.prelude.kint import INT`